libraries/InternalFileSytem/src/flash/flash_cache.c

```diff
@@ -22,12 +22,69 @@
  * THE SOFTWARE.
  */
 
+#include <FreeRTOS.h>
+#include <semphr.h> // This ties to FreeRTOS ... used to serialize flash access
+
+#include "flash_nrf5x.h"
+#include "flash_cache.h"
+#include "nrf_sdm.h"
+#include "nrf_soc.h"
+#include "delay.h"
+#include "rtos.h"
+
+
 #include <string.h>
 #include "flash_cache.h"
 #include "common_func.h"
 #include "variant.h"
 #include "wiring_digital.h"
 
+
+
+static volatile SemaphoreHandle_t _serializeFlashAccess = NULL;
+
+int  _internal_flash_cache_write (flash_cache_t* fc, uint32_t dst, void const * src, uint32_t len);
+int  _internal_flash_cache_read  (flash_cache_t* fc, void* dst, uint32_t addr, uint32_t count);
+void _internal_flash_cache_flush (flash_cache_t* fc);
+
+static inline void _internal_EnsureFlashCacheSemaphoreInitialized()
```

Comment:
I am doing mostly testing and tweaking so far. Everything is good, though this seems to be over complicated. I will add an Static semaphor struct, it will guarantee an mutex creation is always succeeded in my PR.